libIP2Location/IP2Location.c

```diff
@@ -1432,58 +1423,59 @@ char *IP2Location_read_string(FILE *handle, uint32_t position)
 
 float IP2Location_readFloat(FILE *handle, uint32_t position)
 {
+	int endian = 1;
 	float ret = 0.0;
 	uint8_t *cache_shm = memory_pointer;
 	size_t temp;
 
-#if defined(_SUN_) || defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || defined(__powerpc64__)
-	char *p = (char *) &ret;
+	if (*(char*)&endian==0) {    /* BIG ENDIAN */
```

Comment:
Only test is changed, other lines are only indentation.
(such runtime test will be optimized by compiler as always true/false)